src/main/java/com/xmslyz/luhn/logic/LuhnChecksumComputing.java

```diff
@@ -8,18 +8,17 @@ public class LuhnChecksumComputing implements Computation {
 
     @Override
     public String compute(String input) {
-        try {
-            this.number = input.trim().replace(" ", "");
-            setArray(number.length());
-            multiplyWages();
-            countTotal();
-            getChecksum();
-            return this.checkSum;
-        } catch (NumberFormatException | NullPointerException e) {
-            System.out.println("Input error.");
+        if (input == null) throw new NullPointerException("Expected not-null input");
+        if (input.isEmpty()) throw new IllegalArgumentException("Expected empty string");
```

Comment:
W wiadomości wyjątku podajesz co poszło nie tak. Tutaj string jest pusty, a nie chcesz, żeby był pusty, więc wiadomość powinna brzemieć: "oczekuję niepustego stringu".
```suggestion
        if (input.isEmpty()) throw new IllegalArgumentException("Expected non-empty string");
```